### PR TITLE
Adding `prompt` builtin for selecting or customizing the Ion prompt

### PR DESCRIPTION
**Pull request by stratact**
This PR is **work-in-progress** and of need in guidance to get things working together as a builtin.

### CODE REVIEW

**Comment by stratact** [comment]
About the colors part, could I also have a preset of color values, such as `blue`, `green`, `red`, `yellow` etc for the end user to choose from if they don't want to use hex because it might be too hard for them to do. I'd say the hexdecimal option is more of a power user option.

**Comment by mmstick** [comment]
@stratact You can display the list from the colors module.

**Comment by stratact** [comment]
Ah fantastic

**Comment by mmstick** [comment]
(and hexadecimal should be simple since it's literally just `0xRRGGBB`

**Comment by stratact** [comment]
I'm not excluding the hexadecimal idea, it's just that there should be some options that are easier to use and get started with as users become more familiar with ion.

File: src/lib/builtins/prompt.rs [hunk 1]
```diff
@@ -43,16 +48,25 @@ pub(crate) fn prompt(args: &[String], shell: &mut Shell) -> Result<(), String> {
...
+                            Some('c') => { 
+                                match input_iter.next() {
+                                    Some('1') => stdout.ask_color_for("${PWD}", &mut new_prompt),
+                                    Some('2') => stdout.ask_color_for("${SWD}", &mut new_prompt),
+                                    _ => continue,
+                                }
+                            }
+                            Some('d') => { 
+                                unimplemented!();
+                                stdout.ask_color_for(r#"$(top -bn1 | grep "Cpu(s)" | sed "s/.*, *\([0-9.]*\)%* id.*/\1/" | awk '{print 100 - $1"%"}')"#, &mut new_prompt);
```
Comment (stratact) [inline src/lib/builtins/prompt.rs]: I need help figuring out to do an Ionic way of doing `$(top -bn1 | grep "Cpu(s)" | sed "s/.*, *\([0-9.]*\)%* id.*/\1/" | awk '{print 100 - $1"%"}')`.

**Comment by stratact** [comment]
Closing for now, gonna work on this in our GitLab server instead.

**Comment by FriedPandaFries** [comment]
Hello, in ion's current state would it be possible to use something like this <a href="https://github.com/jD91mZM2/powerline-rs">Rust rewrite of Powerline</a>?

**Comment by zen3ger** [comment]
@FriedPandaFries it should. In `~/.config/ion/initrc` define:

```rust
fn PROMPT
    powerline-rs --shell bare $?
end
```

**NOTE**: The redox project moved to [gitlab](https://gitlab.redox-os.org/redox-os/ion), so for further questions either visit the forum/chat or the gitlab issues!

**Pull request closed without merging**